types/next/document.d.ts

```diff
@@ -89,5 +89,5 @@ export class NextScript extends React.Component<NextScriptProps> {}
 export default class Document<IP = {}, C = NextDocumentContext> extends React.Component<
     IP & DocumentProps
 > {
-    getInitialProps(context: C): Promise<IP> | IP;
+    static getInitialProps(context: C): Promise<IP> | IP;
```

Comment:
Yeah, static methods cannot use the instance params. I'll have another look later today as well.

Reply:
Also, can you add some test with your use case?